data/core/src/commonMain/kotlin/com/tunjid/heron/data/network/models/PostConversions.kt

```diff
@@ -82,7 +90,183 @@ internal fun PostView.post(
     viewingProfileId: ProfileId?,
 ): Post {
     val postEntity = postEntity()
-    return post(
+    val quotedPostEntity = quotedPostEntity()
+    val quotedPostProfileEntity = quotedPostProfileEntity()
+
+    val embeddedRecord = if (quotedPostEntity != null && quotedPostProfileEntity != null) {
+        buildPost(
+            postEntity = quotedPostEntity,
+            profileEntity = quotedPostProfileEntity,
+            embeds = quotedPostEmbedEntities(),
+            viewerStatisticsEntity = null,
+            labels = emptyList(),
+            embeddedRecord = null,
+            quote = null,
+        )
+    } else {
+        // Check the record union and extract the embedded types
+        when (val embed = embed) {
+            is PostViewEmbedUnion.RecordView -> {
```

Comment:
This would be a lot neater if you extracted the methods to:

```kotlin
fun FeedGeneratorView.asExternalModel(): FeedGenerator = ...

fun ListView.asExternalModel(): FeedList = ...
```

and so on. You should also do the same foe labels as the same conversion exists for feed generators and lists.